managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1447,8 +1447,17 @@ public void operationFailed(ManagedLedgerException exception) {
 
     @Override
     public void delete(final Position position) throws InterruptedException, ManagedLedgerException {
-        checkNotNull(position);
```

Comment:
We should have these checks in ManagedCursorImpl.java:1524 (asyncDelete)

Reply:
It would fail anyway the cast: `PositionImpl position  = (PositionImpl) pos;`

Reply:
What about the checkNotNull?

Reply:
Yes, the cast will fail if pos is null.

Reply:
(throwing NPE)